src/lib/SaferNames/Name.hs

```diff
@@ -13,44 +13,62 @@
 {-# LANGUAGE TypeFamilies #-}
```

Comment:
I gave up on reviewing this file, because 1,8k lines is a little much 😛  Let me know if there's anything you'd like me to take a look at